File: examples/java/src/main/java/com/example/hello/DateTimeHelloWorld.java
package com.example.hello;

import java.time.ZonedDateTime;

public class DateTimeHelloWorld
{

  public final String message;

  public final ZonedDateTime dateTime;

  public DateTimeHelloWorld(String message, ZonedDateTime dateTime) {

    this.message = message;
    this.dateTime = dateTime;

  }

  public void printMessage() {
    System.out.println("Your Message : " + message + "\nI was given the specific time of " + dateTime + ".");
  }
}

File: examples/cli/src/main/scala/com/example/hello/DateTimeHelloWorld.scala
package com.example.hello

import java.time.ZonedDateTime

class DateTimeHelloWorld(message: String, val dateTime: ZonedDateTime)
{
  def printMessage() = {
    println(s"Your Message : $message\nI was given the specific time of $dateTime.")
  }
}

object DateTimeHelloWorld {
  def apply(message: String, dateTime: ZonedDateTime) = new DateTimeHelloWorld(message, dateTime)
}
